### PR TITLE
VueJS: Added Bootstrap-Vue Form Validation

### PR DESCRIPTION
**Pull request by ali-kha**
New form validation for list page inputs. 
Submit should be disabled out by default. 
> image  

Upon enterting a key, submit should become enabled.
> image  

If entering a key then removing it (so there is 0 chars in the box) error should appear and submit should be grayed out again:
> image  

Submitting an item should disable submit again
> image

### CODE REVIEW

File: templates/Web/Pages/Vue.List/src/components/ListForm.vue [hunk 1]
```diff
@@ -1,25 +1,59 @@
...
+        class="form-control"
+        placeholder="Add text here..."
+        type="text"
+        @input="$emit('input', $event.target.value)"
+      />
+      <b-alert
+        variant="danger"
+        :show="!this.isValid"
+        dismissible
+      >Input must be at least 1 character long.</b-alert>
```
Comment (SahilTara) [inline templates/Web/Pages/Vue.List/src/components/ListForm.vue]: Do you think it would be good to change the constant message and use the const here?

Comment (ali-kha) [reply templates/Web/Pages/Vue.List/src/components/ListForm.vue]: I did the same thing as angular, look at #764

**Review by jsondoo — DISMISSED** [review]
behaviour seems good :)

File: templates/Web/Pages/Vue.List/src/components/ListForm.vue [hunk 2]
```diff
@@ -1,25 +1,57 @@
...
+    });
+    this.$on("onAddListItem", () => {
+      this.isValid = true;
+      this.isSubmitable = false;
+    });
   }
 };
 </script>
+
+<style scoped>
```
Comment (jsondoo) [inline templates/Web/Pages/Vue.List/src/components/ListForm.vue:52]: very cool

File: templates/Web/Pages/Vue.List/src/components/ListForm.vue [hunk 3]
```diff
@@ -1,25 +1,57 @@
...
+        :value="value"
+        aria-label="Add text here..."
+        class="form-control"
+        placeholder="Add text here..."
+        type="text"
+        @input="$emit('input', $event.target.value)"
+      />
+      <b-alert :show="!this.isValid" variant="danger" dismissible>{{this.emptyError}}</b-alert>
+    </div>
+    <span class="nput-group-btn col-1">
```
Comment (SahilTara) [inline templates/Web/Pages/Vue.List/src/components/ListForm.vue]: Think this is a typo correct me if I'm wrong. Should be "input-group-btn"

**Review by jcoc611-microsoft — APPROVED** [review]

File: templates/Web/_composition/Vue/Page.Vue.AddConstantsList/src/constants$wts.Page.Vue.List_gpostaction.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ CONSTANTS.ERROR_MESSAGE = {};
 CONSTANTS.ERROR_MESSAGE.LIST_DELETE = "Request to delete list item failed:";
 CONSTANTS.ERROR_MESSAGE.LIST_ADD = "Request to add list item failed:";
 CONSTANTS.ERROR_MESSAGE.LIST_GET = "Request to get list items failed:";
-CONSTANTS.ERROR_MESSAGE.LIST_EMPTY_MESSAGE = "Please enter a valid message";
+CONSTANTS.ERROR_MESSAGE.LIST_EMPTY_MESSAGE = "Input must be at least 1 character long.";
```
Comment (streetcoder123) [inline templates/Web/_composition/Vue/Page.Vue.AddConstantsList/src/constants$wts.Page.Vue.List_gpostaction.js:9]: Should we change this for React and Angular as well?

Comment (ali-kha) [reply templates/Web/_composition/Vue/Page.Vue.AddConstantsList/src/constants$wts.Page.Vue.List_gpostaction.js:9]: Angular hard coded the value, so I'm pretty sure it wont use this constant. And yes we probably should do this for React, but in a different PR.

**Review by streetcoder123 — APPROVED** [review]

**Review by jsondoo — APPROVED** [review]

**Pull request merged**